backend/utils/sheet.js

```diff
@@ -266,49 +266,73 @@ const getInstitutes_ = async () => {
   return getTableData(institutesList, rawData);
 };
 
-const deleteFromSheet = async (rowIndex, table) => {
-  try {
-    const auth = new google.auth.GoogleAuth({
-      keyFile: SERVICE_ACCOUNT_SECRET_FILE,
-      scopes: "https://www.googleapis.com/auth/spreadsheets",
-    });
-    const authClientObject = await auth.getClient();
-    const googleSheetsInstance = google.sheets({
-      version: "v4",
-      auth: authClientObject,
-    });
+const sendmail = require("../mail");
+const fs = require('fs').promises;
 
-    const spreadsheet = await googleSheetsInstance.spreadsheets.get({
-      spreadsheetId: SPREADSHEET_ID,
-    });
-    const sheets = spreadsheet.data.sheets;
-    const sheet = sheets.find(
-      (s) => s.properties.title === tableSheetMap[table]
-    );
-    const sheetId = sheet.properties.sheetId;
-
-    const result = await googleSheetsInstance.spreadsheets.batchUpdate({
-      spreadsheetId: SPREADSHEET_ID,
-      resource: {
-        requests: [
-          {
-            deleteDimension: {
-              range: {
-                sheetId: sheetId,
-                dimension: "ROWS",
-                startIndex: rowIndex,
-                endIndex: rowIndex + 1,
+const deleteFromSheet = async (rowIndex, table) => {
+  // console.log(table);
+  const result = getUsers_();
+  result.then(value => {
+  const email = value.rows[rowIndex - 1]['User Email'];
+    (async () => {
```

Comment:
Why are we using this IIFE here? The main function is already an async function

Reply:
Right, but in order for the function to execute, result must be processed, and it no longer remains truly asynchronous. In order to keep the mail sending and modification of the sheets separated and asynchronous with each other, I have wrapped it around an async function.

Reply:
So I am guessing you want email sending and sheet modification to go in parallel. For that the right approach is that you collect all promises from these functions and wait till all of them are resolved. Take a look [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)